src/utils.jl

```diff
@@ -184,7 +184,30 @@ end
 macro ogrerr(code, message)
     return quote
         if $(esc(code)) != GDAL.OGRERR_NONE
-            error($message)
+            # find the detailed error message
+            if $(esc(code)) == GDAL.OGRERR_NOT_ENOUGH_DATA
+                detailmsg = "Not enough data."
```

Comment:
Can we make it `detailmsg = if ...` so that we are not defining new variables inside the if statement?

Reply:
I always forget you can do this in Julia - done.